docs/articles/ecomm-recys.md

```diff
@@ -0,0 +1,456 @@
+
+In this article, we'll walk you through how to use the Superlinked library to create an effective RecSys - specifically an e-commerce site that sells mainly clothing - that can be updated in real-time using feedback loops defined by user interactions.
+
+## Achieving personalization despite RecSys vector embedding challenges
+
+While vector embeddings can vastly improve recommendation systems, effectively implementing them requires addressing several challenges, including:
+- **Quality and relevance**: The embedding generation process, architecture, and data must be carefully considered.
+- **Sparse and noisy data**: Embeddings are less effective when they have incomplete or noisy input. Sparse data is the crux of the cold-start problem.
+- **Scalability**: Efficient methods for large datasets are needed; otherwise latency will be an issue.
+
+With Superlinked, you can address these challenges by combining all available data about users and products into rich multimodal vectors. In our e-commerce RecSys example below, we do this using the following Superlinked library elements:
+- **min_max number spaces**: for understanding customer reviews and pricing information
+- **text-similarity space**: for semantic understanding of product information
+- **Events schema** and **Effects** to modify vectors
+- **query time weights** - to define how you want the data to be treated when you run the query, letting you optimize and scale without re-embedding the whole dataset (latency).
+
+By embedding our initially sparse user-specific data (the user's initial product preference), we can handle the cold start problem. As user behavior accrues, we can go much further, *hyper*-personalizing recommendations by embedding this event data, creating a feedback loop that lets you update vectors with user preferences in real time. In addition, Superlinked's query time weights let you fine-tune your retrieval results bias results to match specific user preferences.
+
+Let's get started.
+
+## Building an e-commerce recommendation engine with Superlinked
+
+At the start of development, we have the following product data:
+- number of reviewers
+- product ratings
+- textual description
+- name of product (usually containing brand name)
+- category
+  
+We have two users (user_1 and user_2) whom we can differentiate by either:
+- which of three products offered during registration they chose, or
+- more general characteristics explained in the below paragraph (price, reviews)
+ 
+Users have preferences in textual characteristics of products (description, name, category), and (according to classical economics) ceteris paribus prefer products that:
+- cost less
+- have a lot of reviews
+- have higher ratings
+
```

Comment:
@morkapronczay Do you mean we *know* this from user behavior (event data), or we *assume* this based on classical economics?

Reply:
@morkapronczay see above question.

Reply:
@morkapronczay see above question.

Reply:
yeah common sense or classical economics :)